app/Http/Controllers/DemandController.php

```diff
@@ -0,0 +1,334 @@
+
+        $adpayService = $this->container->has('adpay') ? $this->container->get('adpay') : null;
+        $adpayService instanceof Adpay;
+
+        if ($adpayService) {
+            $adpayService->addEvents([
+              $log->getAdpayJson(),
+          ]);
+        }
+
+        $response = new Response($url);
+
+        // last click id will be used to track conversions
+        $response->headers->setCookie(new Cookie('cid', $request->query->get('cid'), new \DateTime('+ 1 month')));
+
+        $response->setContent(sprintf('<!DOCTYPE html>
+<html>
+  <head>
+      <meta charset="UTF-8" />
+      <meta http-equiv="refresh" content="5;url=%1$s" />
+
+      <title>Redirecting to %1$s</title>
+  </head>
+  <body>
+      Redirecting to <a href="%1$s">%1$s</a>.
+  </body>
+</html>', htmlspecialchars($url, ENT_QUOTES, 'UTF-8')));
+
+        return $response;
+    }
+
+    public function view(Request $request, $id)
+    {
+        if ($request->query->get('r')) {
+            $url = $request->query->get('r');
+            $request->query->remove('r');
+        }
+
+        $logIp = bin2hex(inet_pton($request->getClientIp()));
+
+        $cid = Utils::getRawTrackingId($request->query->get('cid'));
+        $pid = $request->query->get('pid');
+        $tid = Utils::getRawTrackingId($request->cookies->get('tid')) ?: $logIp;
+        $payTo = $request->query->get('pto');
+
+        $keywords = json_decode(Utils::UrlSafeBase64Decode($request->query->get('k')), true);
+
+        $lid = $request->query->get('lid');
+        if ($lid) {
+            $log = EventLog::find($lid);
+        }
+        if (empty($log)) {
+            $log = new EventLog();
+        }
+        $log->publisher_event_id = $pid;
+
+        $log->cid = $cid;
+        $log->banner_id = $id;
+        $log->pay_to = $payTo;
+        $log->tid = $tid;
+        $log->ip = $logIp;
+        $log->their_context = Utils::getImpressionContext($request);
+        $log->event_type = "view";
+        $log->their_userdata=$keywords;
+
+        $log->save();
+
+        $aduser_endpoint = config('app.aduser_endpoint');
+
+        if ($aduser_endpoint) {
+            $iid = $request->query->get('iid') ?: Utils::createTrackingId($this->getParameter('secret'));
+            $backUrl = route('log-keywords', [  'iid' => $iid,
+              'log_id' => $log->id,
+              'r' => $url
+            ]);
+
+            $response = new RedirectResponse($aduser_endpoint . '/setimg/' . $iid . '?r='. Utils::UrlSafeBase64Encode($backUrl));
+        } else {
+            throw new Exception('ADAPY');
+
+            $response = new Response();
+
+            //transparent 1px gif
+            $response->setContent(base64_decode('R0lGODlhAQABAIABAP///wAAACH5BAEKAAEALAAAAAABAAEAAAICTAEAOw=='));
+            $response->headers->set('Content-Type', 'image/gif');
+
+            $adpayService = $this->container->has('adpay') ? $this->container->get('adpay') : null;
+            $adpayService instanceof Adpay;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove or refactor this statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=php%3AS905)